dynamic/discovery/discovery.go

```diff
@@ -32,128 +32,150 @@ import (
+	// name
 	//
 	// NOTE:
-	//	This map of registry works well, since the registry
-	// itself is a map of resources anchored by resource name.
-	// Hence, this combination of apiVersion and resource name
-	// is sufficient to find a particular API resource.
-	resources map[string]apiResourceRegistry
+	//	This can be used to mock GetForAPIVersionResource during
+	// unit test
+	GetForAPIVersionResourceFn func(apiVersion, resource string) *APIResource
 
-	// Client to discover API resource
-	Client discovery.DiscoveryInterface
+	mutex sync.RWMutex
 
-	// GetByResourceFn is a functional type to get API resource
-	// based out of apiVersion & resource name
+	// discovered resources anchored by **apiVersion**
 	//
 	// NOTE:
-	//	This can be used to mock GetByResource during unit test
-	GetByResourceFn func(apiVersion string, resource string) *APIResource
+	//	This map of registry works well, since the registry
+	// itself is a map of discovered resources anchored by
+	// resource name. Hence, this combination of apiVersion
+	// and resource name is sufficient to find a particular
+	// API resource.
+	discoveredResources map[string]apiResourceRegistry
 
 	stopCh, doneCh chan struct{}
 }
 
-// NewAPIResourceManager returns a new instance of
-// APIResourceManager
-func NewAPIResourceManager(c discovery.DiscoveryInterface) *APIResourceManager {
-	return &APIResourceManager{Client: c}
+// NewAPIResourceDiscoverer returns a new instance of APIResourceDiscoverer
+func NewAPIResourceDiscoverer(client discovery.DiscoveryInterface) *APIResourceDiscovery {
+	return &APIResourceDiscovery{
+		DiscoveryClient: client,
+	}
 }
 
-// GetByResource returns the API resource based on the provided
+// GetForAPIVersionResource returns the API resource based on the provided
 // api version and resource
 //
 // NOTE:
 //	resource implies the name of resource which is also the plural
 // notation of kind
-func (mgr *APIResourceManager) GetByResource(apiVersion, resource string) *APIResource {
-	if mgr.GetByResourceFn != nil {
-		return mgr.GetByResourceFn(apiVersion, resource)
+func (d *APIResourceDiscovery) GetForAPIVersionResource(
+	apiVersion string,
+	resource string,
+) *APIResource {
+	if d.GetForAPIVersionResourceFn != nil {
+		return d.GetForAPIVersionResourceFn(apiVersion, resource)
 	}
-	return mgr.getByResource(apiVersion, resource)
+	return d.getForAPIVersionResource(apiVersion, resource)
 }
 
-// getByResource returns the API resource based on the provided
-// api version and resource
+// getForAPIVersionResource returns the discovered API resource
+// corresponding to the provided api version and resource
 //
 // NOTE:
 //	resource implies the name of resource which is also the plural
 // notation of kind
-func (mgr *APIResourceManager) getByResource(apiVersion, resource string) *APIResource {
-	mgr.mutex.RLock()
-	defer mgr.mutex.RUnlock()
-
-	registry, ok := mgr.resources[apiVersion]
+func (d *APIResourceDiscovery) getForAPIVersionResource(
```

Comment:
dum question of begginer - whats the difference between capitalized method ant this one ?

Reply:
It is a good question @grzesuav .
Capitalized methods are public in Go. These can be accessed from other packages.

Reply:
capitalised methods, and global constants and variables are public and accessible in the code importing this package.